reader.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2023 by Posit Software, PBC
+package rsf
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+type rsfReader struct {
+	pos int
+}
+
+func NewReader() Reader {
```

Comment:
Interestingly, the [Go style guide](https://google.github.io/styleguide/go/best-practices#naming) recommends against this

> For functions, do not [repeat the name of the package](https://google.github.io/styleguide/go/decisions#repetitive-with-package).

```go
// Bad:
package yamlconfig

func ParseYAMLConfig(input string) (*Config, error)

// Good:
package yamlconfig

func Parse(input string) (*Config, error)
```

Reply:
Interesting, that makes sense. My thought was it is a specific implementation of the `Reader` interface. But I guess using it would end up looking like `rsf.NewReader()`, which is descriptive enough as it is. Thanks for the link!